server/src/catalog.rs

```diff
@@ -65,147 +101,206 @@ impl Catalog {
+            ));
+            info!("Found table {}.", path);
+        } else if Self::is_path_a_model_table(&dir_entry.path()) {
+            model_table_metadata.push(ModelTableMetadata::try_new(
+                normalized_file_or_folder_name, // A folder for model tables.
+                path.to_string(),
+                &model_table_legacy_segment_file_schema,
+            )?);
+            info!("Found model table {}.", path);
+        } else {
+            error!("File or folder is not a table or model table {}.", path);
         }
-    } else {
-        error!("Unable to open data folder {}.", &data_folder);
+        Ok(())
     }
-    Catalog {
-        table_metadata,
-        model_table_metadata,
-    }
-}
 
-/** Private Methods **/
-// TODO: check the files for tables and model tables have the correct schema.
-fn is_dir_entry_a_table(dir_entry: &DirEntry) -> bool {
-    if let Ok(metadata) = dir_entry.metadata() {
-        if metadata.is_file() {
-            is_dir_entry_a_parquet_file(dir_entry)
-        } else if metadata.is_dir() {
-            if let Ok(mut data_folder) = read_dir(dir_entry.path()) {
-                data_folder
-                    .all(|result| result.is_ok() && is_dir_entry_a_parquet_file(&result.unwrap()))
+    /// Return `true` if `dir_entry` is a readable table, otherwise `false`.
+    fn is_dir_entry_a_table(dir_entry: &DirEntry) -> bool {
+        if let Ok(metadata) = dir_entry.metadata() {
+            if metadata.is_file() {
+                Self::is_dir_entry_a_parquet_file(dir_entry)
+            } else if metadata.is_dir() {
+                if let Ok(mut data_folder) = read_dir(dir_entry.path()) {
+                    data_folder.all(|result| {
+                        result.is_ok() && Self::is_dir_entry_a_parquet_file(&result.unwrap())
+                    })
+                } else {
+                    false
+                }
             } else {
                 false
             }
         } else {
             false
         }
-    } else {
-        false
     }
-}
-
-fn is_dir_entry_a_parquet_file(dir_entry: &DirEntry) -> bool {
-    let mut file = File::open(dir_entry.path()).unwrap();
-    let mut magic_bytes = vec![0u8; 4];
-    let _ = file.read_exact(&mut magic_bytes);
-    magic_bytes == [80, 65, 82, 49] //Magic bytes PAR1
-}
-
-fn new_table_metadata(file_name: String, path: String) -> TableMetadata {
-    let name = if let Some(index) = file_name.find('.') {
-        file_name[0..index].to_string()
-    } else {
-        file_name
-    };
 
-    TableMetadata { name, path }
-}
+    /// Return `true` if `dir_entry` is an readable Apache Parquet file,
+    /// otherwise `false`.
+    fn is_dir_entry_a_parquet_file(dir_entry: &DirEntry) -> bool {
+        if let Ok(mut file) = File::open(dir_entry.path()) {
+            let mut magic_bytes = vec![0u8; 4];
+            let _ = file.read_exact(&mut magic_bytes);
+            magic_bytes == [80, 65, 82, 49] // Magic bytes PAR1.
```

Comment:
The signature is different because it is written in hex on Wikipedia and in binary in the code. I have added a link to the Wikipedia page as a doc comment in the code.